### PR TITLE
tests: add two schematics test

### PR DESCRIPTION
**Pull request by jor2**
### Description
Added schematics test for two other tests in pr_test.go

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: tests/pr_test.go [hunk 1]
```diff
@@ -367,3 +299,76 @@ func TestRunOverride(t *testing.T) {
...
+		Tags:                   []string{"test-schematic"},
+		DeleteWorkspaceOnFail:  false,
+		WaitJobCompleteMinutes: 60,
+	})
+
+	options.TerraformVars = []testschematic.TestSchematicTerraformVar{
+		{Name: "ibmcloud_api_key", Value: options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], DataType: "string", Secure: true},
+		{Name: "region", Value: options.Region, DataType: "string"},
+		{Name: "prefix", Value: options.Prefix, DataType: "string"},
+		{Name: "ssh_key", Value: sshPublicKey(t), DataType: "string"},
```
Comment (ocofaigh) [inline tests/pr_test.go:343]: An ssh key should only be passed to the VSI and VSI quickstart patterns. For VSI standard, the var name is `ssh_public_key` and for quickstart it is `ssh_key`

File: tests/pr_test.go [hunk 2]
```diff
@@ -367,3 +299,76 @@ func TestRunOverride(t *testing.T) {
...
+}
+
+func TestRunVSIPatternSchematics(t *testing.T) {
+	t.Parallel()
+
+	options := setupOptionsSchematics(t, "rok-sc", vsiPatternTerraformDir)
+
+	err := options.RunSchematicTest()
+	assert.NoError(t, err, "Schematic Test had unexpected error")
+}
```
Comment (ocofaigh) [inline tests/pr_test.go:374]: Missing `TestRunVPCPatternSchematics` which should replace the `TestRunVpcPattern` test

File: tests/pr_test.go [hunk 3]
```diff
@@ -367,3 +299,76 @@ func TestRunOverride(t *testing.T) {
...
+	options := setupOptionsSchematics(t, "rok-sc", roksPatternTerraformDir)
+
+	err := options.RunSchematicTest()
+	assert.NoError(t, err, "Schematic Test had unexpected error")
+}
+
+func TestRunVSIPatternSchematics(t *testing.T) {
+	t.Parallel()
+
+	options := setupOptionsSchematics(t, "rok-sc", vsiPatternTerraformDir)
```
Comment (ocofaigh) [inline tests/pr_test.go:370]: `rok-sc` -> `vsi-sc`

File: tests/pr_test.go [hunk 4]
```diff
@@ -367,3 +299,76 @@ func TestRunOverride(t *testing.T) {
...
+		{Name: "ibmcloud_api_key", Value: options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], DataType: "string", Secure: true},
+		{Name: "region", Value: options.Region, DataType: "string"},
+		{Name: "prefix", Value: options.Prefix, DataType: "string"},
+		{Name: "ssh_key", Value: sshPublicKey(t), DataType: "string"},
+	}
+
+	return options
+}
+
+func TestRunQuickStartPatternSchematics(t *testing.T) {
```
Comment (ocofaigh) [inline tests/pr_test.go:349]: `TestRunQuickStartPatternSchematics` -> `TestRunVSIQuickStartPatternSchematics`

File: tests/pr_test.go [hunk 5]
```diff
@@ -367,3 +299,76 @@ func TestRunOverride(t *testing.T) {
...
+		TemplateFolder:         dir,
+		Prefix:                 "ocp-sc",
+		Tags:                   []string{"test-schematic"},
+		DeleteWorkspaceOnFail:  false,
+		WaitJobCompleteMinutes: 60,
+	})
+
+	options.TerraformVars = []testschematic.TestSchematicTerraformVar{
+		{Name: "ibmcloud_api_key", Value: options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], DataType: "string", Secure: true},
+		{Name: "region", Value: options.Region, DataType: "string"},
```
Comment (ocofaigh) [inline tests/pr_test.go:341]: can you set `add_atracker_route` to `false` here please?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Comment by ocofaigh** [comment]
Looks like some backend issue `The system is currently unable to fulfill the request. Please retry. If the failure persists, contact support and provide the trace value from this error.`

Retrying..

**Comment by ocofaigh** [comment]
`TestRunUpgradeRoksPattern` failed with a timeout when provisioning VPN gateway: `timeout while waiting for state to become 'done, ' (last state: 'provisioning', timeout: 10m0s)`

`TestRunRoksPatternSchematics` failed in the schematics destroy, however the destroy actually passed. I saw the same thing yesterday and have reported to schematics team

`TestRunVSIQuickStartPatternSchematics` failed on apply with `Error: cannot add more than 5 gateways to the selected region` - @toddgiguere Is the schematics tests also selecting a region based on VPC availability? I wonder should we disable transit gateway for all but one of of tests for SLZ?

**Comment by toddgiguere** [comment]
@ocofaigh The tests are using region selection correctly, however they are all running in parallel and we are not using a shared cloudInfoService, so my guess is that with the timing they are all selecting the same region.

If we have some limitations then we should probably adjust to using a shared cloudInfo so that different regions are selected for all tests.

@jor2 I can explain how this works if needed.

**Comment by jor2** [comment]
> @ocofaigh The tests are using region selection correctly, however they are all running in parallel and we are not using a shared cloudInfoService, so my guess is that with the timing they are all selecting the same region.
> 
> 
> 
> If we have some limitations then we should probably adjust to using a shared cloudInfo so that different regions are selected for all tests.
> 
> 
> 
> @jor2 I can explain how this works if needed.

Fixed worked, thanks :)

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**